Source/ARTPushActivationStateMachine.m

```diff
@@ -255,6 +256,7 @@ - (void)deviceUpdateRegistration:(ARTErrorInfo *)error {
         }
     } error:nil];
     [request setValue:[[_rest defaultEncoder] mimeType] forHTTPHeaderField:@"Content-Type"];
+    [request setDeviceAuthentication:local.id localDevice:local];
```

Comment:
Ok, I understand that's a bit tricky in that particular case because it's using the local Device Id. But you have other cases where the Device Id is passed by the lib user, so `[request setDeviceAuthentication:argDeviceId localDevice:localDevice]` can happen. Even so, I will add a convenience method when the Device Id is the local device, something like `[request setDeviceAuthentication:local]`.